powersimdata/network/constants/region/usa.py

```diff
@@ -0,0 +1,239 @@
+        "NC",
+        "SC",
+        "GA",
+        "FL",
+        "AL",
+        "MS",
+        "TN",
+        "KY",
+        "WV",
+        "OH",
+        "MI",
+        "IN",
+        "IL",
+        "WI",
+        "MN",
+        "IA",
+        "MO",
+        "AR",
+        "LA",
+        "OK",
+        "KS",
+        "NE",
+        "SD",
+        "ND",
+    },
+    "ERCOT": {"TX"},
+    "Western": {"WA", "OR", "CA", "NV", "AZ", "UT", "NM", "CO", "WY", "ID", "MT"},
+}
+
+cb = [i for j in range(2, 4) for i in combinations(["ERCOT", "Eastern", "Western"], j)]
+for c in cb:
+    interconnect2abv[interconnect_to_name(c)] = {
+        a for i in c for a in interconnect2abv[i]
+    }
+
+
+def get_interconnect_mapping(zone, model):
+    """Return interconnect mapping.
+
+    :param pandas.DataFrame zone: information on zones of a grid model.
+    :param str model: the grid model.
+    :return: (*dict*) -- mappings of interconnect to other areas.
+    """
+
+    def _substitute(entry):
+        return {
+            i: ast.literal_eval(repr(entry).replace("ERCOT", sub))[i]
+            for i in mapping["interconnect"]
+        }
+
+    mapping = dict()
+    sub = "Texas" if model == "usa_tamu" else "ERCOT"
+
+    name = interconnect_to_name(zone["interconnect"].replace(sub, "ERCOT").unique())
+
+    mapping["interconnect"] = ast.literal_eval(
+        repr(name2component[name]).replace("ERCOT", sub)
+    ) | {name}
+
+    mapping["name2interconnect"] = _substitute(name2interconnect)
+    mapping["name2component"] = _substitute(name2component)
+    mapping["interconnect2timezone"] = _substitute(interconnect2timezone)
+    mapping["interconnect2abv"] = _substitute(interconnect2abv)
+    mapping["interconnect2loadzone"] = {
+        i: set(zone.set_index("abv")["zone_name"].loc[list(a)])
+        for i, a in mapping["interconnect2abv"].items()
+    }
+    mapping["interconnect2id"] = {
+        i: set(zone.reset_index().set_index("abv")["zone_id"].loc[list(a)])
+        for i, a in mapping["interconnect2abv"].items()
+    }
+    return mapping
+
+
+def get_state_mapping(zone):
+    """Return state mapping.
+
+    :param pandas.DataFrame zone: information on zones of a grid model.
+    :return: (*dict*) -- mappings of states to other areas.
+    """
+    mapping = dict()
+
+    mapping["state"] = set(zone["state"])
+    mapping["abv"] = set(zone["abv"])
+    mapping["state2loadzone"] = {
+        k: set(v)
+        for k, v in zone.groupby("state")["zone_name"].unique().to_dict().items()
+    }
+    mapping["abv2loadzone"] = {
+        k: set(v)
+        for k, v in zone.groupby("abv")["zone_name"].unique().to_dict().items()
+    }
+    mapping["abv2id"] = {k: set(v) for k, v in zone.groupby("abv").groups.items()}
+    mapping["id2abv"] = {k: v for k, v in zone["abv"].to_dict().items()}
+    mapping["state2abv"] = {k: v for k, v in zip(zone["state"], zone["abv"])}
+    mapping["abv2state"] = {k: v for k, v in zip(zone["abv"], zone["state"])}
```

Comment:
You can also do e.g. `mapping["state2abv"] = dict(zip(zone["state"], zone["abv"]))`.

Reply:
Done